### PR TITLE
Update to Bevy 0.13

### PR DESCRIPTION
**Pull request by VirxEC**

### CODE REVIEW

File: Cargo.toml
Comment (JonahPlusPlus) [inline Cargo.toml:1]: The patch versions will need to be removed. Only the major and minor versions are needed.

File: src/lib.rs [hunk 1]
```diff
@@ -177,11 +198,11 @@ fn spectator_update(
             let delta_lateral = (right - left) * speed;
             let delta_vertical = (up - down) * speed;
 
-            let mut forward = camera_transform.forward();
+            let mut forward = Vec3::from(camera_transform.forward());
```
Comment (JonahPlusPlus) [inline src/lib.rs]: This can just be a dereference (I think? `*camera_transform.forward()`).

**Review by JonahPlusPlus — CHANGES_REQUESTED** [review]
If you could just change the bevy versions to exclude the patch version, I'll merge this.

(Removing `Vec3::from` and adding the dereference would be nice for code quality, but not necessary)

**Comment by VirxEC** [comment]
Done

**Review by JonahPlusPlus — APPROVED** [review]

File: src/lib.rs [hunk 2]
```diff
@@ -146,20 +146,41 @@ fn spectator_update(
...
+                1f32
+            } else {
+                0f32
+            };
+
+            let left = if keys.pressed(KeyCode::KeyA) {
+                1f32
+            } else {
+                0f32
+            };
```
Comment (JasmineLowen) [inline src/lib.rs:171]: This is repeating the same pattern quiet often. You could use `condition as usize as f32` for an easy condition but I think that's a bit implicit so maybe something like

```rust
fn one_if_true(condition: bool) { if condition { 1.0 } else { 0.0 } }

let backward = one_if_true(keys.pressed(KeyCode::KeyS));
let right = one_if_true(keys.pressed(KeyCode::KeyD));
let left = one_if_true(keys.pressed(KeyCode::KeyA));
let forward = one_if_true(keys.pressed(KeyCode::KeyW));
```

would be more clean?

File: examples/3d_scene.rs [hunk 3]
```diff
@@ -23,17 +23,14 @@ fn setup(
...
-        material: materials.add(Color::rgb(0.3, 0.5, 0.3).into()),
+        mesh: meshes.add(Rectangle::from_size(Vec2::splat(5.0))),
+        material: materials.add(Color::rgb(0.3, 0.5, 0.3)),
         ..default()
     });
     // cube
     commands.spawn(PbrBundle {
-        mesh: meshes.add(Mesh::from(shape::Cube { size: 1.0 })),
-        material: materials.add(Color::rgb(0.8, 0.7, 0.6).into()),
+        mesh: meshes.add(Cuboid::from_size(Vec3::splat(1.0))),
```
Comment (JasmineLowen) [inline examples/3d_scene.rs]: Vec3::splat(1.0) is the default Value, so you could use `Cuboid::default()` here as well

**Review by JasmineLowen — APPROVED** [review]
Honestly, just reviewing to give this more relevancy again.

All of my review comments should be considered non-blocking, feel free to ignore.

Overall, clean update 👍🏼

File: src/lib.rs (hunk 2 above)
Comment (Shatur) [reply src/lib.rs:171]: I personally prefer the current implementation.

**Pull request merged**